txpool/src/containers/dependency.rs

```diff
@@ -0,0 +1,482 @@
+                                    .as_ref()
+                                    .expect("contract origin to be present");
+                                check.push(*origin.tx_id());
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn check_if_coin_input_can_spend_output(
+        output: &Output,
+        input: &Input,
+        is_output_filled: bool,
+    ) -> anyhow::Result<()> {
+        if let Input::Coin {
+            owner,
+            amount,
+            color,
+            ..
+        } = input
+        {
+            let i_owner = owner;
+            let i_amount = amount;
+            let i_color = color;
+            match output {
+                Output::Coin { to, amount, color } => {
+                    if to != i_owner {
+                        return Err(Error::NotInsertedIOWrongOwner.into());
+                    }
+                    if amount < i_amount {
+                        return Err(Error::NotInsertedIOWrongAmount.into());
+                    }
+                    if color != i_color {
+                        return Err(Error::NotInsertedIOWrongAssetIt.into());
+                    }
+                }
+                Output::Contract { .. } => return Err(Error::NotInsertedIOConractOutput.into()),
+                Output::Withdrawal { .. } => {
+                    return Err(Error::NotInsertedIOWithdrawalInput.into());
+                }
+                Output::Change { to, color, amount } => {
+                    if to != i_owner {
+                        return Err(Error::NotInsertedIOWrongOwner.into());
+                    }
+                    if color != i_color {
+                        return Err(Error::NotInsertedIOWrongAssetIt.into());
+                    }
+                    if is_output_filled && amount < i_amount {
+                        return Err(Error::NotInsertedIOWrongAmount.into());
+                    }
+                }
+                Output::Variable { to, amount, color } => {
+                    if is_output_filled {
+                        if to != i_owner {
+                            return Err(Error::NotInsertedIOWrongOwner.into());
+                        }
+                        if amount < i_amount {
+                            return Err(Error::NotInsertedIOWrongAmount.into());
+                        }
+                        if color != i_color {
+                            return Err(Error::NotInsertedIOWrongAssetIt.into());
+                        }
+                    }
+                    // else do nothing, everything is variable and can be only check on execution
+                }
+                Output::ContractCreated { .. } => {
+                    return Err(Error::NotInsertedIOConractOutput.into())
+                }
+            };
+        } else {
+            panic!("Use it only for coin output check");
```

Comment:
Why panic instead of just returning a normal `Error`?

Reply:
If this uses function named `check_if_coin_input_can_spend_output` is used for checking `coin_contract` than this means somebody is missusing it